controllers/machinehealthcheck_targets.go

```diff
@@ -116,16 +117,22 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 
 	// the node has not been set yet
 	if t.Node == nil {
-		// status not updated yet
-		if t.Machine.Status.LastUpdated == nil {
-			return false, timeoutForMachineToHaveNode
+		// TODO change to checking ControlPlaneReadyCondition in v1alpha4, when cluster.spec.controlPlaneRef will be required.
+		// We can't do this yet because ControlPlaneReadyCondition is only set when you're using a control plane provider,
+		// and that is optional in v1alpha3.
+		if !conditions.Has(t.Cluster, clusterv1.InfrastructureReadyCondition) || conditions.IsFalse(t.Cluster, clusterv1.InfrastructureReadyCondition) {
+			// Cluster infrastructure is not ready yet, return a nextCheck time of 0 because we'll get requeued when
+			// the Cluster is updated.
+			return false, 0
 		}
-		if t.Machine.Status.LastUpdated.Add(timeoutForMachineToHaveNode).Before(now) {
+
+		infraReadyTime := conditions.GetLastTransitionTime(t.Cluster, clusterv1.InfrastructureReadyCondition)
+		if infraReadyTime.Add(timeoutForMachineToHaveNode).Before(now) {
```

Comment:
How would this affect Machines that were provisioned long after the Cluster infrastructure is initialized, should this be updated to compare both the time since infraReadyTime and the machine last updated time against unhealthy duration to account for both cases? 

Alternatively, do we need a net new condition on a machine to indicate that provisioning has been started that can accurately be checked against rather than the approximations we are currently using?

Reply:
Yeah I need to take that into account. This is on hold until I file a new issue about adding a condition for control plane available.